examples/tv-casting-app/linux/simple-app-helper.cpp

```diff
@@ -168,22 +175,75 @@ void SubscribeToMediaPlaybackCurrentState(matter::casting::memory::Strong<matter
         kMinIntervalFloorSeconds, kMaxIntervalCeilingSeconds);
 }
 
+CHIP_ERROR InitCommissionableDataProvider(LinuxCommissionableDataProvider & provider, LinuxDeviceOptions & options)
+{
+    ChipLogProgress(Discovery, "InitCommissionableDataProvider()");
+    chip::Optional<uint32_t> setupPasscode;
+
+    if (options.payload.setUpPINCode != 0)
+    {
+        setupPasscode.SetValue(options.payload.setUpPINCode);
+        ChipLogProgress(Discovery, "InitCommissionableDataProvider() using setupPasscode: %d", setupPasscode.Value());
+    }
+    else if (!options.spake2pVerifier.HasValue())
+    {
+        uint32_t defaultTestPasscode = 0;
+        chip::DeviceLayer::TestOnlyCommissionableDataProvider TestOnlyCommissionableDataProvider;
+        VerifyOrDie(TestOnlyCommissionableDataProvider.GetSetupPasscode(defaultTestPasscode) == CHIP_NO_ERROR);
+
+        ChipLogError(Support,
+                     "*** WARNING: Using temporary passcode %u due to no neither --passcode or --spake2p-verifier-base64 "
+                     "given on command line. This is temporary and will be deprecated. Please update your scripts "
+                     "to explicitly configure onboarding credentials. ***",
+                     static_cast<unsigned>(defaultTestPasscode));
+        setupPasscode.SetValue(defaultTestPasscode);
+        options.payload.setUpPINCode = defaultTestPasscode;
+    }
+    else
+    {
+        // Passcode is 0, so will be ignored, and verifier will take over. Onboarding payload
+        // printed for debug will be invalid, but if the onboarding payload had been given
+        // properly to the commissioner later, PASE will succeed.
+    }
+
+    // Default to the minimum PBKDF iterations (1000) for this example implementation. For TV devices and TV casting app production
```

Comment:
does increasing the number in the example have a downside? Seems like this is easy for a developer to miss.

Reply:
Increasing the number of iterations will increase the computational time required to derive the key. This can slow down the authentication process, especially on devices with limited processing power. For a TV app running on a Raspberry Pi 4, 1,000 Iterations might take ~20 ms and the max allowed 100,000 iterations might take ~2 seconds.

Ideally the developers should measure the actual performance on the target device and pick a number that works best for their use case.
I'll make it a clear *** WARNING: log and add some notes about performance.